README.md

```diff
@@ -1,149 +1,194 @@
+```
+$ cd /path/to/code
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --generate-config
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep
+```
 
-#### sgrep Registry
+## Configuring
 
-[r2c](https://r2c.dev) provides a [registry](https://github.com/returntocorp/sgrep-rules) of config files tuned using our [analysis platform](https://app.r2c.dev) on thousands of repositories. To use:
+For simple patterns use the `--lang` and `--pattern` flags. This mode of
+operation is useful for quickly iterating on a pattern on a single file or
+folder:
 
-```bash
-sgrep --config r2c
+```
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --lang javascript --pattern 'eval(...)' path/to/file.js
 ```
 
-### Default
-
-Default configs are loaded from `.sgrep.yml` or multiple files matching `.sgrep/**/*.yml` and can be overridden by using `--config <file|folder|yaml_url|tarball_url|registy_name>`
+To fine-tune your searching, specify the `--help` flag:
 
-## Design
+```
+$ docker run --rm returntocorp/sgrep --help
+```
 
-Sgrep has a design philosophy that emphasizes simplicity and a single pattern being as expressive as possible:
+### Configuration Files
 
-1. **Use concrete code syntax:** easy to learn
-2. **Metavariables \($X\)**: abstract away code
-3. **'...' operator:** abstract away sequences
-4. **Knows about code equivalences:** one pattern can match many equivalent variations on the code
-5. **Less is more:** abstract away additional details
+For advanced configuration use the `--config` flag. This flag automagically
+handles a multitude of input types:
 
-## Patterns
+* `--config <file|folder|yaml_url|tarball_url|registy_name>`
 
-Patterns are snippets of code with variables and other operators that will be parsed into an AST for that language and will be used to search for that pattern in code. See [docs/config/simple.md](docs/config/simple.md) for full documentation.
+In the absense of this flag, a default configuration is loaded from `.sgrep.yml`
+or multiple files matching `.sgrep/**/*.yml`.
 
-### Metavariables
+#### Operators
 
-`$X`, `$FOO`, `$RETURNCODE` are all examples of metavariables. You can referance them later in your pattern and `sgrep` will ensure they match. **Metavariables can only contain uppercase ASCII characters**; `$x` and `$SOME_VALUE` are not valid metavariables.
+Configuration files make use of two primary operators:
+
+* **Metavariables like $X, $LIST, or $METAVARIABLE.** Metavariable names can
+only contain uppercase characters. Metavariables are used to track a variable
+across a specific code scope.
+* **The '...' (splat) operator.** The splat operator abstracts away sequences
+so you don't have to sweat the details of a particular code pattern.
+
+Let's consider an example:
+
+```yaml
+rules:
+  - id: open-never-closed
+    patterns:
+      - pattern: $FD = open(...)
+      - pattern-not-inside: |
```

Comment:
Since we're explaining $X and ..., it might be a leap to have people grok `pattern-not-inside` at the same time.  Is there another example we could use?

Reply:
Good catch here. My thoughts were that we need: 1.) an obvious code smell that people can connect with (e.g. files opened but never closed), 2.) a simple, short example that makes use of metavariables and the ellipsis operator. This was just the first one that came to mind. I think people should be able to grok `pattern-not-inside` as a "negation" operator, but I'm open to other ideas too. Do you have an example in mind that meets the above criteria?

Reply:
🤔 What about modifying a list while iterating over it? https://sgrep.live/5K0. We could either briefly explain `pattern-either` because I think it's easier to understand than `pattern-not-inside`, or we could just grab one of those patterns. Probably `.append`

Reply:
I think, at the very least, we need to explain what `pattern`, etc. are and do (maybe up at line 115?) before introducing this example.